### PR TITLE
Laravel modular

### LINKED ISSUE #14: [BE] Set up a modular folder structure by feature
### Task Requirements

Laravel offers a generic folder structure by type which is not flexible enough and will cause chaos after a while.

I recommend that we use a modular approach to the project so that we can manage the code easily.

Investigate about modular monoliths and check out packages like [InterNACHI/modular](https://github.com/InterNACHI/modular) and [nwidart/laravel-modules](https://nwidart.com/laravel-modules/v6/introduction) and implement the better option.

We will have at least 3 main modules at the start:
- Website - everything related to the website will be here (blog, contact, newsletter etc.)
- Platform - The members area
- Admin - The admin area

### PR DESCRIPTION
**Pull request by nikoladomazetovikj**
This pull request avoids issues from migrating modular structure under the `app` folder in PR #26

### CODE REVIEW

File: app/platform/tests/PlatformServiceProviderTest.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace App\Platform\Tests;
+
+class PlatformServiceProviderTest extends TestCase
```
Comment (davorminchorov) [inline app/platform/tests/PlatformServiceProviderTest.php]: You can remove this file, we won't need it.

File: README.md [hunk 2]
```diff
@@ -48,6 +48,22 @@ Access the documentation:
...
+Initially our system has 3 modules:
+- `Website` - everything related to the website will be here (blog, contact, newsletter etc.)
+- `Platform` - The members area
+- `Admin` - The admin area
+
+To work with `artisan` commands for each module check the documentation:
+
+https://github.com/InterNACHI/modular?tab=readme-ov-file#commands
+
+Note: All modules are placed under `app-modules` folder
```
Comment (davorminchorov) [inline README.md]: This needs to be `app` instead of `app-modules`

**Review by davorminchorov — COMMENTED** [review]
The names of the modules should always be using Capital letters.

Other than that, it looks good, thanks for all of these changes.

**Review by davorminchorov — APPROVED** [review]

File: app/Platform/src/Providers/PlatformServiceProvider.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Platform\Providers;
+
+use Illuminate\Support\ServiceProvider;
+
+class PlatformServiceProvider extends ServiceProvider
+{
+	public function register()
```
Comment (bzrncev) [inline app/Platform/src/Providers/PlatformServiceProvider.php:9]: Do we need this file at all for now? We can easily add it later if needed. If you plan to keep it, you can remove register/boot

File: app/Website/routes/website-routes.php [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<?php
```
Comment (bzrncev) [inline app/Website/routes/website-routes.php:1]: should we keep the same naming, like web/api.php here as well?

File: app/Website/src/Providers/WebsiteServiceProvider.php [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Website\Providers;
+
+use Illuminate\Support\ServiceProvider;
+
+class WebsiteServiceProvider extends ServiceProvider
+{
+	public function register()
```
Comment (bzrncev) [inline app/Website/src/Providers/WebsiteServiceProvider.php:9]: Same as above here

**Review by bzrncev — APPROVED** [review]

**Comment by davorminchorov** [comment]
I tested this locally and it works great, it will do the job just fine.

Please fix the merge conflicts so that we can merge it and start using it.

**Comment by nikoladomazetovikj** [comment]
@davorminchorov  all conflicts are resolved

**Pull request merged**